src/MeshBVH.js

```diff
@@ -7,50 +7,48 @@ export default class MeshBVH {
 
 	constructor( geo, options = {} ) {
 
-		// default options
-		options = Object.assign( {
-
-			strategy: CENTER,
-			maxDepth: 40,
-			maxLeafTris: 10,
-			verbose: true
-
-		}, options );
-		options.strategy = Math.max( 0, Math.min( 2, options.strategy ) );
-
 		if ( geo.isBufferGeometry ) {
 
-			this._roots = this._buildTree( geo, options );
+			// default options
+			options = Object.assign( {
 
-		} else {
+				strategy: CENTER,
+				maxDepth: 40,
+				maxLeafTris: 10,
+				verbose: true,
+				index: null
 
-			throw new Error( 'MeshBVH: Only BufferGeometries are supported.' );
+			}, options );
+			options.strategy = Math.max( 0, Math.min( 2, options.strategy ) );
 
-		}
+			if ( ! options.index ) {
 
-	}
+				const triCount = geo.attributes.position.count / 3;
+				const indexCount = triCount * 3;
+				const indexArray = new ( triCount > 65535 ? Uint32Array : Uint16Array )( indexCount );
 
-	/* Private Functions */
+				for ( let i = 0; i < indexCount; i ++ ) {
 
-	_ensureIndex( geo ) {
+					indexArray[ i ] = i;
 
-		if ( ! geo.index ) {
+				}
 
-			const triCount = geo.attributes.position.count / 3;
-			const indexCount = triCount * 3;
-			const index = new ( triCount > 65535 ? Uint32Array : Uint16Array )( indexCount );
-			geo.setIndex( new THREE.BufferAttribute( index, 1 ) );
+				options.index = new THREE.BufferAttribute( indexArray, 1, true );
```

Comment:
It's a little weird to mutate the options they passed in, in my opinion, but I don't care very much.